src/iOS/Xamarin.iOS/Samples/Augmented Reality/NavigateAR/RoutePlanner.cs

```diff
@@ -0,0 +1,364 @@
+            };
+
+            // Add the views.
+            View.AddSubviews(_mapView, _helpLabel, toolbar);
+
+            // Lay out the views.
+            NSLayoutConstraint.ActivateConstraints(new[]{
+                _mapView.TopAnchor.ConstraintEqualTo(View.TopAnchor),
+                _mapView.BottomAnchor.ConstraintEqualTo(View.BottomAnchor),
+                _mapView.LeadingAnchor.ConstraintEqualTo(View.LeadingAnchor),
+                _mapView.TrailingAnchor.ConstraintEqualTo(View.TrailingAnchor),
+                _helpLabel.TopAnchor.ConstraintEqualTo(View.SafeAreaLayoutGuide.TopAnchor),
+                _helpLabel.LeadingAnchor.ConstraintEqualTo(View.LeadingAnchor),
+                _helpLabel.TrailingAnchor.ConstraintEqualTo(View.TrailingAnchor),
+                _helpLabel.HeightAnchor.ConstraintEqualTo(40),
+                toolbar.LeadingAnchor.ConstraintEqualTo(View.LeadingAnchor),
+                toolbar.TrailingAnchor.ConstraintEqualTo(View.TrailingAnchor),
+                toolbar.BottomAnchor.ConstraintEqualTo(View.SafeAreaLayoutGuide.BottomAnchor)
+            });
+        }
+
+        public override void ViewDidLoad()
+        {
+            base.ViewDidLoad();
+            Initialize();
+        }
+
+        private async void Initialize()
+        {
+            // Create and add the map.
+            _mapView.Map = new Map(Basemap.CreateImagery());
+
+            try
+            {
+                // Configure location display.
+                _mapView.LocationDisplay.AutoPanMode = LocationDisplayAutoPanMode.Recenter;
+                await _mapView.LocationDisplay.DataSource.StartAsync();
+                _mapView.LocationDisplay.IsEnabled = true;
+
+                // Enable authentication.
+                SetOAuthInfo();
+
+                // Create the route task.
+                _routeTask = await RouteTask.CreateAsync(new System.Uri("https://route.arcgis.com/arcgis/rest/services/World/Route/NAServer/Route_World"));
+
+                // Create route display overlay and symbology.
+                _routeOverlay = new GraphicsOverlay();
+                SimpleLineSymbol routeSymbol = new SimpleLineSymbol(SimpleLineSymbolStyle.Solid, System.Drawing.Color.Yellow, 1);
+                _routeOverlay.Renderer = new SimpleRenderer(routeSymbol);
+
+                // Create stop display overlay.
+                _stopsOverlay = new GraphicsOverlay();
+
+                // Add the overlays to the map.
+                _mapView.GraphicsOverlays.Add(_routeOverlay);
+                _mapView.GraphicsOverlays.Add(_stopsOverlay);
+
+                // Wait for the user to place stops.
+                _mapView.GeoViewTapped += MapView_GeoViewTapped;
+
+                // Updat the help text.
+                _helpLabel.Text = "Tap to set a start point";
+            }
+            catch (Exception ex)
+            {
+                ShowMessage("Failed to start sample", "Error");
```

Comment:
new UIAlertView("Error", "Failed to start sample", (IUIAlertViewDelegate) null, "OK", null).Show();